qiling/os/posix/syscall/socket.py

```diff
@@ -78,12 +78,24 @@ def ql_syscall_socket(ql, socket_domain, socket_type, socket_protocol, *args, **
         if idx == -1:
             regreturn = -1
         else:
-            if ql.verbose >= QL_VERBOSE.DEBUG: # set REUSEADDR options under debug mode
+            # ql_socket.open should use host platform based socket_type.
+            try:
+                emu_socket_value = socket_type
+                emu_socket_type = socket_type_mapping(socket_type, ql.archtype)
+                socket_type = getattr(socket, emu_socket_type)
+                ql.log.info("Convert emu_socket_type {}:{} to host platform based socket_type {}:{}".format(
+                    emu_socket_type, emu_socket_value, emu_socket_type, socket_type))
+
+            except Exception:
```

Comment:
I think it must convert.
even a x8664 linux to x8664linux, so we should remove try and except

Reply:
I use this try and except to capture `getattr` exception. Maybe change it like below?
```python
           try:
                emu_socket_value = socket_type
                emu_socket_type = socket_type_mapping(socket_type, ql.archtype)
                socket_type = getattr(socket, emu_socket_type)
                ql.log.info("Convert emu_socket_type {}:{} to host platform based socket_type {}:{}".format(
                    emu_socket_type, emu_socket_value, emu_socket_type, socket_type))

            except AttributeError:
                ql.log.error("Can't convert emu_socket_type {}:{} to host platform based socket_type".format(
                    emu_socket_type, emu_socket_value))
                raise

            except Exception:
                ql.log.error("Can't convert emu_socket_type {} to host platform based socket_type".format(
                    emu_socket_value))
                raise
```

Reply:
agree. we should raise an error here.